hpedockerplugin/file_manager.py

```diff
@@ -0,0 +1,479 @@
+            LOG.info("Initializing 3PAR driver...")
+            self._primary_driver = self._initialize_driver(
+                host_config, self.src_bkend_config, self.tgt_bkend_config)
+
+            self._hpeplugin_driver = self._primary_driver
+            LOG.info("Initialized 3PAR driver!")
+        except Exception as ex:
+            msg = "Failed to initialize 3PAR driver for array: %s!" \
+                  "Exception: %s"\
+                  % (self.src_bkend_config.hpe3par_api_url,
+                     six.text_type(ex))
+            LOG.info(msg)
+            raise exception.HPEPluginStartPluginException(
+                reason=msg)
+
+        # If replication enabled, then initialize secondary driver
+        if self.tgt_bkend_config:
+            LOG.info("Replication enabled!")
+            try:
+                LOG.info("Initializing 3PAR driver for remote array...")
+                self._remote_driver = self._initialize_driver(
+                    host_config, self.tgt_bkend_config,
+                    self.src_bkend_config)
+            except Exception as ex:
+                msg = "Failed to initialize 3PAR driver for remote array %s!" \
+                      "Exception: %s"\
+                      % (self.tgt_bkend_config.hpe3par_api_url,
+                         six.text_type(ex))
+                LOG.info(msg)
+                raise exception.HPEPluginStartPluginException(reason=msg)
+
+        # self._initialize_default_metadata()
+
+    def get_backend(self):
+        return self._backend
+
+    def get_mediator(self):
+        return self._hpeplugin_driver
+
+    def get_file_etcd(self):
+        return self._fp_etcd_client
+
+    def get_etcd(self):
+        return self._etcd
+
+    def get_config(self):
+        return self._hpepluginconfig
+
+    # Create metadata for the backend if it doesn't exist
+    def _initialize_default_metadata(self):
+        try:
+            metadata = self._fp_etcd_client.get_backend_metadata(self._backend)
+        except exception.EtcdBackendMetadataDoesNotExist:
+            metadata = {
+                'cpg_fpg_map': {
+                    'used_ips': [],
+                    'counter': 0,
+                    'default_fpgs': {self.src_bkend_config.hpe3par_cpg: None}
+                }
+            }
+            self._fp_etcd_client.save_backend_metadata(metadata)
+
+    def _initialize_configuration(self):
+        self.src_bkend_config = self._get_src_bkend_config()
+        self.tgt_bkend_config = None
+
+    def _get_src_bkend_config(self):
+        LOG.info("Getting source backend configuration...")
+        hpeconf = self._hpepluginconfig
+        config = acp.ArrayConnectionParams()
+        for key in hpeconf.keys():
+            value = getattr(hpeconf, key)
+            config.__setattr__(key, value)
+
+        LOG.info("Got source backend configuration!")
+        return config
+
+    def _initialize_driver(self, host_config, src_config, tgt_config):
```

Comment:
@imran-ansari This has unused parameter tgt_config, please remove it.

Reply:
Now that replication is not required, we can remove it. Will take care of it.